staging/src/k8s.io/apiserver/pkg/storage/testing/store_tests.go

```diff
@@ -2759,3 +2759,312 @@ func RunTestListPaging(ctx context.Context, t *testing.T, store storage.Interfac
 		t.Errorf("unexpected items: %#v", names)
 	}
 }
+
+func RunTestNamespaceScopedList(ctx context.Context, t *testing.T, store storage.Interface) {
+	tests := []struct {
+		name               string
+		requestedNamespace string
+		recursive          bool
+		fieldSelector      fields.Selector
+		indexFields        []string
+		gotPods            []example.Pod
+		expectPods         []example.Pod
+	}{
+		{
+			name:          "request without namespace, without field selector",
+			recursive:     true,
+			fieldSelector: fields.Everything(),
+			gotPods: []example.Pod{
+				*baseNamespacedPod("t1-foo1", "t1-ns1"),
```

Comment:
Same comment as I had about hardcoding namespaces.
I very much prefer this pattern:
https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/storage/testing/watcher_tests.go#L1456

In this case fieldSelector would probably have to be a function taking namespace as a param, but it's still better imho.

Reply:
Right, but here we cannot easily isolate test, because we use namespace as selector. I'm worried that adding some automatic suffixes just obscures the test, as what is  presented testcases is different than what will be shown in errors.

Reply:
It's not about adding obscure suffixes - it's about making it parameters. So basically switching to:
```
fieldSelector: func(namespace string) fields.Selector
inputPods: []func(namespace string) example.Pod
expectPods: []func(namespace string) example.Pod
````

Reply:
Done